packages/node-experimental/src/sdk/transaction.ts

```diff
@@ -0,0 +1,62 @@
+import type { Hub } from '@sentry/core';
+import { sampleTransaction, Transaction } from '@sentry/core';
+import type {
+  ClientOptions,
+  CustomSamplingContext,
+  Hub as HubInterface,
+  Scope,
+  TransactionContext,
+} from '@sentry/types';
+import { uuid4 } from '@sentry/utils';
+
+/**
+ * This is a fork of core's tracing/hubextensions.ts _startTransaction,
+ * with some OTEL specifics.
+ */
+export function startTransaction(
+  hub: HubInterface,
+  transactionContext: TransactionContext,
+  customSamplingContext?: CustomSamplingContext,
+): Transaction {
+  const client = hub.getClient();
+  const options: Partial<ClientOptions> = (client && client.getOptions()) || {};
+
+  let transaction = new NodeExperimentalTransaction(transactionContext, hub as Hub);
+  transaction = sampleTransaction(transaction, options, {
+    parentSampled: transactionContext.parentSampled,
+    transactionContext,
+    ...customSamplingContext,
+  });
+  if (transaction.sampled) {
+    transaction.initSpanRecorder(options._experiments && (options._experiments.maxSpans as number));
+  }
+  if (client && client.emit) {
+    client.emit('startTransaction', transaction);
+  }
+  return transaction;
+}
+
+/**
+ * This is a fork of the base Transaction with OTEL specific stuff added.
+ */
+export class NodeExperimentalTransaction extends Transaction {
```

Comment:
l: iiuc, `Transaction::finish` shouldn't do anything anymore, right? Should we override it here to print a warning that it noops if it's called? I guess chances are low that users would call `finish` given that they'll only work with spans, so feel free to disregard.

Reply:
actually users should not get a hold of a transaction ever 😅 `startTransaction` is not exported, and only used in the span exporter.